Ternary Search Tree/TernarySearchTree.swift

```diff
@@ -33,52 +33,52 @@ public class TernarySearchTree<Element> {
 
     - returns: Value indicating insertion success/failure.
     */
-    public func insert(data: Element, withKey key: String) -> Bool {
-        return insertNode(&root, withData: data, andKey: key, atIndex: 0)
+    @discardableResult public func insert(data: Element, withKey key: String) -> Bool {
+        return insert(node: &root, withData: data, andKey: key, atIndex: 0)
     }
 
     /**
      Helper method for insertion that does the actual legwork. Insertion is performed recursively.
 
-     - parameter aNode:     The current node to insert below.
+     - parameter node:     The current node to insert below.
      - parameter data:      The data being inserted.
      - parameter key:       The key being used to find an insertion location for the given data
      - parameter charIndex: The index of the character in the key string to use to for the next node.
 
      - returns: Value indicating insertion success/failure.
      */
-    private func insertNode(inout aNode: TSTNode<Element>?, withData data: Element, andKey key: String, atIndex charIndex: Int) -> Bool {
+    private func insert(node: inout TSTNode<Element>?, withData data: Element, andKey key: String, atIndex charIndex: Int) -> Bool {
 
         //sanity check.
         if key.characters.count == 0 {
             return false
         }
 
         //create a new node if necessary.
-        if aNode == nil {
-            let index = key.startIndex.advancedBy(charIndex)
-            aNode = TSTNode<Element>(key: key[index])
+        if node == nil {
+            let index = key.index(key.startIndex, offsetBy: charIndex)
+            node = TSTNode<Element>(key: key[index])
         }
 
```

Comment:
About this one....I'm having doubts about how to fix this. It seems that central parts of the algorithm is based on creating new nodes if the passed in node is `nil`.

The nodes used are of type `TSTNode`, and they contain three children: `left`, `right` and `middle` and these are then used when we insert a new node in the tree.

The first time you call this method, you do so with the `root` node. That is `nil` at that time, so you create a new value and use that.

Later on, when building your tree, you look at the key to determine whether you should do a recursive call to `insert` with the left, middle or right child node. At first the children are `nil` too, so they are created when `insert` is called.

So...you can probably get around the `nil` check in some way, but I cannot figure out how right now. 

Input is welcome 😄